api/src/controllers/collections.ts

```diff
@@ -1,13 +1,49 @@
 import { Router } from 'express';
+import Busboy from 'busboy';
 import asyncHandler from '../utils/async-handler';
-import { CollectionsService, MetaService } from '../services';
-import { ForbiddenException } from '../exceptions';
+import { CollectionsService, MetaService, XliffService } from '../services';
+import { ForbiddenException, InvalidPayloadException } from '../exceptions';
 import { respond } from '../middleware/respond';
 import { validateBatch } from '../middleware/validate-batch';
 import { Item } from '../types';
+import logger from '../logger';
 
 const router = Router();
 
+const multipartHandler = asyncHandler(async (req, res, next) => {
+	if (req.is('multipart/form-data') === false) return next();
+	const busboy = new Busboy({ headers: req.headers });
+	let fileCount = 0;
+	res.locals.fields = {};
+	busboy.on('field', (fieldname: keyof File, val) => {
+		res.locals.fields[fieldname] = val;
+	});
+	busboy.on('file', async (fieldname, fileStream, filename, encoding, mimetype) => {
+		fileCount++;
+		const { format } = res.locals.fields;
+		if (['xliff', 'xliff2'].includes(format)) {
+			if (fileCount > 1) {
+				busboy.emit('error', new InvalidPayloadException(`Only one import file can be used for XLIFF format.`));
+			}
+			let content = '';
+			fileStream.on('data', (d) => (content += d)).on('end', () => (res.locals.data = content));
+		} else {
```

Comment:
I think we'll spin this into a bigger `/import` endpoint/system, as per https://github.com/directus/directus/issues/2927. I'd rather not handle this under the `/collections` endpoint, as that will cause confusion between the functionality (creating a new collection vs reading the file and injecting the content into the DB)

Reply:
Right now the export/import of XLIFF content happens from parent collection in terms of UI. The functionality available if there is one or more fields with type 'translations' in collection. I was thinking that this is will be the more natural way neither then adding this functionality right to translations collection, but maybe I'm not right. Translators (from my experience) usually operating with the original entities, not the translated items. Also, in case if they will work with exporting from translations collection they will need to make some filtering for specific language before making export.  Now, what if they selected two or more items that related to different languages, so how to export them? We can use exactly the same interface with languages dropdown in order to filter them out, but it will be kind of duplication of UX behavior - they will need to select language twice then, once for filtering items and second time before exporting...

Reply:
> I was thinking that this is will be the more natural way neither then adding this functionality right to translations collection, but maybe I'm not right. 

Yes you're absolutely right from a UX perspective in the app. That being said, I think it's technically easier to reason about if the export itself actually happens on the related translations collections, instead of through the o2m field

Reply:
Are, so you mean to refactor XliffService so it will work with translations collection directly instead of using the parent one as a parameter, right?

Reply:
Yes 👍🏻

Reply:
Ok, will try to do this as well.